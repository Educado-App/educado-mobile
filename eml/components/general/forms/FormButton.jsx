import React from 'react';

// Components
import { TouchableOpacity } from 'react-native';
import { View } from 'react-native';
import Text from '../Text';
import PropTypes from 'prop-types';

/**
 * Button component for eg. login and register screens.
 * @param {Object} props Should contain the following properties:
 * - label: String
 * - onPress: Function
 * @returns {React.Element} Button component
 */
export default function FormButton(props) {

	FormButton.propTypes = {
		children: PropTypes.string,
		disabled: PropTypes.bool,
		onPress: PropTypes.func,
		style: PropTypes.array,
		type: PropTypes.string
	};

	// Put this here for possible custom styling
	const typeStyles = {
		primary: 'bg-primary',
		error: 'bg-error',
		warning: 'bg-yellow',
	};

	return <>
		<View>
			<TouchableOpacity
				className={'px-4 py-4 rounded-medium ' +
					(typeStyles[props.type] ?? typeStyles.primary) +
					(props.disabled ? ' opacity-50' : '')}
				style={props.style ?? null}
				onPress={props.onPress}
				disabled={props.disabled}
			>
				<Text className="text-center font-sans-bold text-body text-projectWhite">
					{props.children}
				</Text>
			</TouchableOpacity>
		</View>
	</>;
}
