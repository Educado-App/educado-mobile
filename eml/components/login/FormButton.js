import React, { useState } from "react";

import { Text, View, TextInput, Pressable } from "react-native";

export default function FormButton(props) {
  return (
    <View className='mx-10'>
      <Pressable 
        className='flex-auto items-center bg-[#5ECCE9] px-[40px] py-4 rounded-[8px] opacity-100 active:opacity-70 active:shadow-inner'
        onPress={props.onPress}
      >
        <Text className='text-white text-[16px]'>{props.label}</Text>
      </Pressable>
    </View>

  )
}