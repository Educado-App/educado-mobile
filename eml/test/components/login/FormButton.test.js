import renderer from "react-test-renderer";
import React from "react";
import FormButton from "../../../components/general/forms/FormButton";

test('Ensure that the button is rendered correctly', async () => {
  let formButton;
  await renderer.act(() => {
    formButton = renderer.create(<FormButton />);
  });
  let tree = formButton.toJSON();
  expect(tree).toMatchSnapshot();
});