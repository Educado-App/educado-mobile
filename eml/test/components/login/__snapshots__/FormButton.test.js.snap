// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ensure that the button is rendered correctly 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#5ECCE9",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Montserrat-Regular",
          },
          {
            "color": "#383838",
          },
          {
            "fontSize": 16,
          },
          [
            {
              "textAlign": "center",
            },
            {
              "fontFamily": "Montserrat-Bold",
            },
            {
              "fontSize": 16,
            },
            {
              "color": "#FFFFFF",
            },
          ],
        ]
      }
    />
  </View>
</View>
`;
